types/lodash/common/object.d.ts

```diff
@@ -1021,8 +1021,41 @@ declare module "../index" {
         functionsIn(): CollectionChain<string>;
     }
 
-    type GetIndexedField<T, K> = K extends keyof T
-        ? T[K]
+    /**
+     * If a value of type T can be indexed by a key of type K
+     * GetIndexedField<T, K> deduces the type of value at that key,
+     * otherwise it returns undefined type.
+     */
+    type GetIndexedField<T, K> =
+        // This covers a lot of ground: indexing arrays and strings by number,
+        // retrieving their "length", indexing objects by their proper keys.
+        K extends keyof T ? T[K]
+
+        // Corner-case: T = Record<string, SomeType>; K = number.
```

Comment:
I think this comment can be simplified a bunch. Maybe instead just give an example, e.g. "handle T[1] the same as T['1']"